File: algorithm_study/test/1041.py
robot1 = list(map(str, input().strip()))
robot2 = list(map(str, input().strip()))
recoderList = []
for r1 in robot1:
    for r2 in robot2:
        if (r2 == r1):
            if r2 not in recoderList:
                recoderList.append(r1)

recoderList.sort()
nums = len(recoderList)
if (nums == 0):
    print("WuXiao", end="")
elif (nums == 1):
    print("ZhiHui")
    print(recoderList[0], end="")
else:
    print("XLuo")
    print(nums)
    for i in range(0, nums):
        if (i < nums - 1):
            print(recoderList[i], end="-")
        else:
            print(recoderList[i], end="")

File: algorithm_study/test/1041.cpp
#include <iostream>
#include <set>
#include <string>

using namespace std;
int main() {
  string robot1, robot2;
  cin >> robot1;
  cin >> robot2;
  set<char> recoderlist;
  for (auto r1 : robot1)
    for (auto r2 : robot2)
      if (r1 == r2)
        recoderlist.insert(r1);
  auto nums = recoderlist.size();
  if (nums == 0)
    cout << "WuXiao";
  else if (nums == 1)
    cout << "ZhiHui" << '\n' << *recoderlist.begin();
  else {
    cout << "XLuo" << '\n';
    cout << nums << '\n';
    int count = 0;
    for (auto it : recoderlist) {
      if (count < nums - 1)
        cout << it << "-";
      else
        cout << it;
      count++;
    }
  }
}